src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -45,13 +45,14 @@ class java_bytecode_convert_classt:public messaget
   {
     add_array_types();
 
-    if(parse_tree.loading_successful)
+    bool loading_success=parse_tree.loading_successful;
+    if(loading_success)
       convert(parse_tree.parsed_class);
-    else if(string_preprocess.is_known_string_type(
-              parse_tree.parsed_class.name))
+
```

Comment:
These changes make me a little bit anxious that the semantics of the code have changed. Before the change, the `if...else if...else` cascading conditional statement means that there is a single set of 3 possible consequence expressions, only 1 of which will be evaluated.

This change has essentially broken the one set (of consequent statements/expressions) into two sets, one with one consequence and the other with 2, both of which can be evaluated independently of each other, just based on their conditions. 

What I mean with this, that only one consequent expression could be evaluated at a point in time (because they were exclusive due to the way the conditional statement was written), now 2 consequent expressions can be evaluated at the same time. Do we want this?